api/envoy/extensions/filters/http/grpc_field_extraction/v3/config.proto

```diff
@@ -0,0 +1,133 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_field_extraction.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_field_extraction.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_field_extraction/v3;grpc_field_extractionv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC Field Extraction]
+// gRPC Field Extraction :ref:`configuration overview
+// <config_http_filters_grpc_field_extraction>`.
+// [#extension: envoy.filters.http.grpc_field_extraction]
+
+// GrpcFieldExtractionConfig supports extracting the fields from first gRPC
+// request message no matter if it is unary or streaming and inject the result
+// in headers. It will **block headers propagation** before the extraction
+// is done, so that the following filters can use the extraction result in headers. No op if isn't a gRPC
```

Comment:
What happens if the field is not found? I assume it's blocking until the entire protobuf message is consumed?

Reply:
The filter will 1) block the propagation 2) try to buffer one complete message 3) extract fields from that message.

If not found, it rejects the requests.